### DISCUSSION TITLE
Are there any tutorials on installing Stripe?

### DISCUSSION THREAD

**Discussion by amityweb** [saleor/saleor#15907] [General] [answered]
I have deployed Saleor via Heroku. I followed some instructions for this some time ago. 

I have Stripe enabled. There are two plugins, one is deprecated the other is active. 

The Stripe tutorials for Saleor are pretty bad, I cannot find any documentation of actually adding or showing the payment form on the site. 

I notice today it says dont use the plugin, use the App version. 

But I have no idea how to install that. The app store buttons are greyed out. I found this one https://github.com/saleor/saleor-app-payment-stripe but the instructions on there are terrible also. I FINALLY found out I need to copy the manifest URL https://saleor-app-payment-stripe.vercel.app/ and into the Install External app. Bu then I get an error:

```
Couldn’t Install Stripe
This app expects to be installed only in allowed Saleor instances
```
I have read something about Saleor Cloud and apps. I am not using Saleor Cloud? Its a self hosting on Heroku. 

So how does one install apps? If we cant install apps why is there an apps section? And what else should I do? Still with the Plugin that was there If thats the case, how does one show/add a payment form? 

Just needs some documentation. its an ecommerce store that has documentation for everything except actually taking payment 😂

Thank you!

**Comment by jor23dan** [comment]
I have get same error in install

**Reply by tomh4** [reply]
Anyone was able to install the app?

**Comment by amityweb** [comment]
I've abandoned Saleor. I was wasting too much time on things like this instead of actually building the website. Other systems do the same so we just switched to something that works out of the box and up and running in an hour. 

It really isn't our job to be trying to make these systems work.

**Reply by jha-r** [reply]
What system are you using and found useful .

**Reply by amityweb** [reply]
Just WooCommerce

**Comment by lkostrowski** [comment, marked as answer]
Hey @amityweb 

Sorry to hear about your struggles. We are aware of gaps in our docs :( We will look closer into that.

To answer your question - Saleor is an open-source platform *and* a SaaS (Saleor Cloud). 
- SaaS version means that we host it - both Saleor and Apps. 
- For an open-source, self-hosted approach, you need to deploy Saleor on your own and deploy apps too.

Apps are separate "microservices" and contrary to plugins, they are not embedded. 
The majority of apps Saleor develops are serverless and can be easily deployed to Vercel (apps are built on top of Next.js)

Currently, the Stripe app is not yet available in the Saleor App Store (we had a beta once, but we put it down). We plan to bring it back once we ensure it's ready

**Comment by amityweb** [comment]
Thanks. We hosted the open source version ourself and its totally not clear how to install apps, it seems the self hosted version is tied into the hosted app store and doesn't work? All a bit strange. 

Either way, its an eCommerce platform without Stripe, one of the most common website payment gateways in the world - to me thats a huge failure. We only use Stripe unless the customer requests an alternative, which is hardly ever at all, so we have Stripe as default in like 99% of projects the past 10 years and its missing from Saleor. I abandoned Saleor because I spent several days or more just trying to take a payment for a product using Stripe which would normally take me under an hour.

**Reply by lkostrowski** [reply]
Thanks for your feedback. We will improvie our docs around installing apps and ensure non-cloud users know what to do.

Regarding Stripe - we fully understand that. We definitely plan to have it one day, but we must consider other priorities as well.

I hope you will give us another chance one day :)

**Reply by serhatopcu** [reply]
@lkostrowski 
I am experiencing similar issues. I haven't even been able to add a single application with Stripe, and I keep encountering the same error every time.


Couldn’t Install BankTransfer
The auth data given during registration request could not be used to fetch app ID. This usually means that App could not connect to Saleor during installation. Saleor URL that App tried to connect: http://localhost:8000/graphql/

**Reply by serhatopcu** [reply]
I think I solved the problem, and maybe this will help those who encounter this message.
First, run Saleor Core on a domain, then define PUBLIC_URL. After that, you won't receive the Couldn’t Install Stripe messages anymore.
